test/workbox-routing/node/Router.mjs

```diff
@@ -0,0 +1,302 @@
+          expect(error.details).to.have.property('funcName').that.eql('registerRoutes');
+          expect(error.details).to.have.property('paramName').that.eql('routes');
+        }
+      );
+
+      await expectError(
+        () => router.registerRoutes([{handle: HANDLER, method: METHOD}]),
+        'missing-a-method',
+        (error) => {
+          expect(error.details).to.have.property('moduleName').that.eql('workbox-routing');
+          expect(error.details).to.have.property('className').that.eql('Router');
+          expect(error.details).to.have.property('funcName').that.eql('registerRoutes');
+          expect(error.details).to.have.property('paramName').that.eql('routes');
+        }
+      );
+
+      await expectError(
+        () => router.registerRoutes([{match: MATCH, method: METHOD}]),
+        'incorrect-type',
+        (error) => {
+          expect(error.details).to.have.property('moduleName').that.eql('workbox-routing');
+          expect(error.details).to.have.property('className').that.eql('Router');
+          expect(error.details).to.have.property('funcName').that.eql('registerRoutes');
+          expect(error.details).to.have.property('paramName').that.eql('routes');
+        }
+      );
+
+      await expectError(
+        () => router.registerRoutes([{match: MATCH, method: METHOD, handler: {}}]),
+        'missing-a-method',
+        (error) => {
+          expect(error.details).to.have.property('moduleName').that.eql('workbox-routing');
+          expect(error.details).to.have.property('className').that.eql('Router');
+          expect(error.details).to.have.property('funcName').that.eql('registerRoutes');
+          expect(error.details).to.have.property('paramName').that.eql('routes');
+        }
+      );
+
+      await expectError(
+        () => router.registerRoutes([{match: MATCH, handle: HANDLER}]),
+        'incorrect-type',
+        (error) => {
+          expect(error.details).to.have.property('moduleName').that.eql('workbox-routing');
+          expect(error.details).to.have.property('className').that.eql('Router');
+          expect(error.details).to.have.property('funcName').that.eql('registerRoutes');
+          expect(error.details).to.have.property('paramName').that.eql('routes');
+        }
+      );
+    });
+
+    it(`should modify the internal arrays of routes when register/unregister is called`, function() {
+      const router = new Router();
+
+      // Routes without an explicit method will default to GET.
+      const getRoute1 = new Route(MATCH, HANDLER);
+      const getRoute2 = new Route(MATCH, HANDLER, 'GET');
+      const putRoute1 = new Route(MATCH, HANDLER, 'PUT');
+      const putRoute2 = new Route(MATCH, HANDLER, 'PUT');
+      // We support passing in Objects that match the expected interface in addition to Routes.
+      const postRoute = {
+        match: MATCH,
+        handler: {handle: HANDLER},
```

Comment:
Can handler just be a function here or does it *have* to be wrapped like this?

Would love it if it didn't need to be wrapped.

Reply:
See https://github.com/GoogleChrome/workbox/issues/848

I'd like to leave as-is for now and then come to a decision on that issue, as changing things will involve a bunch of other cascading changes that will further muddy this PR.